text/0314-cdk-construct-hub.md

```diff
@@ -0,0 +1,714 @@
+solution.
+
+## FAQ
+
+### What are we launching today?
+
+Construct Hub is a new website that makes it easy for developers to discover,
+use, and share constructs. The community has already published hundreds of CDK,
+CDK8s, CDKtf, projen, ... constructs on the public registries like npmjs.com,
+PyPI, Maven Central, and NuGet. However those are distributed repositories,
+managed by numerous organizations, and lack centralized search tools sufficient
+for developers to easily find the cloud infrastructure patterns they are looking
+for, in the programming language they want to use.
+
+Construct Hub automatically indexes all construct libraries published to the
+various general purpose package managers, so that developers have a single place
+to search for constructs, regardless of preferred programming language or
+provisioning engine. Developers will find constructs published by AWS, partners,
+and individual developers. Constuct Hub also generates unified documentation for
+the constructs in all supported programming languages.
+
+Construct Hub is also available as a deploymable CDK construct library, so
+that enterprises and users with specific needs can host a private instance of
+the catalog, making interally developed constructs easier for teams to find.
+Just like the public instance of Construct Hub scans public repositories for
+construct libraries, the private Construct Hub instance can be configured to
+scan private repositories like AWS CodeArtifact, JFrog Artifactory, and others.
+
+### Why should I use this feature?
+
+The Construct Hub is a centralized place that helps discover and share
+custom-made constructs compatible with the AWS Cloud Development Kit (CDK), CDK
+for Kubernetes (CDK8s), CDK for Terraform (CDKtf), and other constructs domains
+such as Projen. It shortens the time it takes to ship cloud applications by
+allowing developers to discover constructs that address their infrastructure
+requirements, and provides a unified cross-language documentation experience to
+help developers integrate those constructs into their application.
+
+Customers can also deploy a private instance of Construct Hub to facilitate
+sharing infrastructure abstractions published to a private package registry
+within the organization.
+
+### Where can I find the Construct Hub?
+
+The public instance of Construct Hub is at: `https://constructs.dev`.
+
+### How do I publish a package to Construct Hub?
+
+Packages published to the public npm registry (npmjs.com) are automatically
+indexed by the public instance of Construct Hub as long as they meet the
+following prerequisites:
+
+1. Cross-language support has been configured correctly using [jsii]
+1. The `cdk` keyword is referenced (in the `keywords` key of `package.json`)
+
+[jsii]: https://aws.github.io/jsii
+
+### Can I opt packages out of the public instance of Construct Hub?
+
+There is currently no self-service way to un-list a package from the Construct
```

Comment:
Can we do this somehow with project metadata, like another keyword in package.json? That would guarantee that it was the project owner who decided to not list the package, or de-list it.

Reply:
Package metadata cannot be updated on already published versions, making it impossible to un-publish existing versions this way...